src/main/java/de/retest/recheck/review/workers/SaveFilterWorker.java

```diff
@@ -1,49 +1,52 @@
 package de.retest.recheck.review.workers;
 
-import java.io.File;
+import static java.nio.file.Files.createDirectories;
+import static java.nio.file.Files.write;
+
 import java.io.IOException;
-import java.io.PrintStream;
 import java.nio.file.Path;
-import java.util.Optional;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
 import java.util.stream.Stream;
 
-import org.apache.commons.io.FileUtils;
-
 import de.retest.recheck.ignore.CacheFilter;
 import de.retest.recheck.ignore.Filter;
-import de.retest.recheck.ignore.JSFilterImpl;
-import de.retest.recheck.ignore.RecheckIgnoreLocator;
+import de.retest.recheck.ignore.PersistentFilter;
 import de.retest.recheck.review.GlobalIgnoreApplier;
 import de.retest.recheck.review.GlobalIgnoreApplier.PersistableGlobalIgnoreApplier;
 import de.retest.recheck.review.ignore.io.Loaders;
+import lombok.extern.slf4j.Slf4j;
 
+@Slf4j
 public class SaveFilterWorker {
 
 	private final GlobalIgnoreApplier applier;
-	private final RecheckIgnoreLocator locator;
 
 	public SaveFilterWorker( final GlobalIgnoreApplier applier ) {
-		this( applier, new RecheckIgnoreLocator() );
-	}
-
-	protected SaveFilterWorker( final GlobalIgnoreApplier applier, final RecheckIgnoreLocator locator ) {
 		this.applier = applier;
-		this.locator = locator;
 	}
 
 	public void save() throws IOException {
-		final Optional<Path> ignorePath = locator.getProjectIgnoreFile();
 		final PersistableGlobalIgnoreApplier persist = applier.persist();
 
 		// Filter JSFilter because that would create unnecessary file content.
-		final Stream<Filter> filters = persist.getIgnores().stream() //
+		final Stream<PersistentFilter> filters = persist.getIgnores().stream() //
 				.map( this::extractCachedFilter ) //
-				.filter( filter -> !(filter instanceof JSFilterImpl) );
-		final Stream<String> save = Loaders.filter().save( filters );
-
-		final File ignoreFile = ignorePath.orElse( locator.getUserIgnoreFile() ).toFile();
-		try ( final PrintStream writer = new PrintStream( FileUtils.openOutputStream( ignoreFile ) ) ) {
-			save.forEach( writer::println );
+				.filter( filter -> (filter instanceof PersistentFilter) ) //
+				.map( f -> (PersistentFilter) f );
```

Comment:
You do not need to do this anymore. Since you now have `List<PersistentFilter`, which never can be a `CacheFilter`.

Instead I would sugggest to change ll 39-40.
```
final Map<Path, List<Filter>> mapped =
filters.collect( Collectors.groupingBy( PersistentFilter::getPath, Collectors.mapping( f-> extractCachedFilter(f.getFilter), Collectors.toList() ) ) );
```